stl/inc/xstring

```diff
@@ -2792,9 +2792,10 @@ public:
         _Container_proxy_ptr<_Alty> _Proxy(_Alproxy, _My_data); // throws
 
         if (_New_capacity < _New_size) {
-            _New_capacity           = _Calculate_growth(_New_size, _BUF_SIZE - 1, max_size());
-            const pointer _Fancyptr = _Getal().allocate(_New_capacity + 1); // throws
+            _New_capacity           = _Calculate_growth(_New_size, _BUF_SIZE - 1, max_size()) + 1;
+            const pointer _Fancyptr = _Allocate_at_least_helper(_Getal(), _New_capacity); // throws
             _Ptr                    = _Unfancy(_Fancyptr);
+            _New_capacity--;

```

Comment:
Style: We conventionally use predecrement unless we specifically need post's behavior.